base/range.jl

```diff
@@ -681,8 +681,8 @@ end
 
 function length(r::AbstractUnitRange{T}) where T
     @_inline_meta
-    a = Integer(last(r) - first(r)) # even when isempty, by construction (with overflow)
-    return a + one(a)
+    a = last(r) - first(r) # even when isempty, by construction (with overflow)
+    return Integer(a + oneunit(a))
```

Comment:
There appears to be a handful of other places this changes is needed (anywhere you see `+one(T)`. Could you do those also?

Reply:
I changed `one(a)` to `oneunit(a)` in the remaining methods (0078f1e). I don’t think it is strictly necessary (the length of a range should always be dimensionless), but it seems more consistent.

Reply:
It looks like the `Integer` cast / call also needs to be moved, for consistency, too?

Reply:
Yes, I now moved those, too.